### ISSUE TITLE
Cloud Trace integration

### ISSUE THREAD

**Issue by dinvlad** [GoogleCloudPlatform/cloudsql-proxy#600]
## Feature Description
We'd like to enable Cloud Trace logs for CloudSQL proxy connections. More specifically, to see which hosts CloudSQL proxy makes connections to, in Cloud Trace logs.

This is important for 2 major reasons:
1. While we could rely on CloudSQL proxy logs for this, not all applications invoke CloudSQL proxy the same way (some might do it via the Docker image, others as a binary embedded in the app, which might not even produce any logs), and parsing of the logs is not always a reliable solution (as the log format might change over time).
2. There's convenience in having all app requests (both HTTP and Cloud SQL proxy) in one place (Cloud Trace), as opposed to having to piece them and to correlate them together from multiple trace/logging services.

Obviously, it would be hard to maintain trace _context_ for this, but at least we could more easily correlate Cloud SQL proxy connections with the application that uses them (by statically assigning an app ID/label as a command-line arg to the proxy).

## Alternatives Considered
I'm unaware of any workarounds for this.

## Additional Context
None

**Comment by enocom** [comment]
Related to https://github.com/GoogleCloudPlatform/cloudsql-proxy/issues/169.

**Comment by NicholasAzar** [comment]
This would be very useful!

**Comment by enocom** [comment]
We're definitely interested in adding support for Cloud Trace, and metrics generally. 

Right now we're waiting on two things before picking up this work:

- the general availability of [OpenTelemetry](https://opentelemetry.io/), and
- some structural improvements to the proxy itself that will make adding metrics more feasible.

I'm going to merge this issue with #169 in the meantime.

**Comment by dinvlad** [comment]
OK, I was just hoping that this FR doesn't get "lost" because in our mind, Cloud Trace is different functionality from Cloud Monitoring integration.

**Comment by enocom** [comment]
In that case, I'm happy to re-open this.

**Comment by enocom** [comment]
I've added an [explicit note](https://github.com/GoogleCloudPlatform/cloudsql-proxy/issues/169#issuecomment-776978018) about support for Cloud Trace on #169. We've already renamed the issue to make it clear that the proxy needs support for metrics and tracing across a number of monitoring options. If you feel like I haven't captured the gist of this issue in #169, feel free to add more context over there -- it's the tracking issue for monitoring and tracing generally.
